compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/CodegenOptimizationUtils.java

```diff
@@ -0,0 +1,57 @@
+/*
+ *  Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.projects.util;
+
+import org.ballerinalang.model.elements.PackageID;
+
+import static io.ballerina.projects.util.CodegenOptimizationConstants.BALLERINAX;
+import static io.ballerina.projects.util.CodegenOptimizationConstants.BALLERINA_JBALLERINA_JAVA;
+import static io.ballerina.projects.util.CodegenOptimizationConstants.BALLERINA_LANG;
+import static io.ballerina.projects.util.CodegenOptimizationConstants.BALLERINA_OBSERVE;
+import static io.ballerina.projects.util.CodegenOptimizationConstants.DOT_DRIVER;
+
+/**
+ * Common utils for codegen optimization implementation.
+ *
+ * @since 2201.10.0
+ */
+public final class CodegenOptimizationUtils {
```

Comment:
Since this is a util class let's give this a private constructor to prevent instantiation

Reply:
Fixed in c8c56fff75d0077964eb8a793986bb68f3e8a915.